src/app/downloads/downloads-table.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Subscription } from 'rxjs/Subscription';
+import { EpisodeMetricsModel, EpisodeModel, FilterModel, INTERVAL_DAILY, INTERVAL_HOURLY, INTERVAL_15MIN } from '../ngrx/model';
+import { selectEpisodes, selectFilter, selectEpisodeMetrics } from '../ngrx/reducers';
+import { filterPodcastMetrics, filterAllPodcastEpisodes, filterEpisodeMetrics, metricsData, getTotal } from '../shared/util/metrics.util';
+import { mapMetricsToTimeseriesData, dayMonthDate, hourlyDateFormat } from '../shared/util/chart.util';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'metrics-downloads-table',
+  template: `
+    <table *ngIf="episodeTableData && episodeTableData.length">
+      <thead>
+        <tr>
+          <th class="sticky">Episode</th>
+          <th>Release Date</th>
+          <th>Total for period</th>
+          <th *ngFor="let date of dateRange">{{date}}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr *ngFor="let episode of episodeTableData">
+          <td class="sticky">{{episode.title}}</td>
+          <td>{{episode.releaseDate}}</td>
+          <td>{{episode.totalForPeriod}}</td>
+          <td *ngFor="let download of episode.downloads">{{download.value}}</td>
+        </tr>
+      </tbody>
+    </table>
+  `,
+  styleUrls: ['downloads-table.component.css']
+
+})
+export class DownloadsTableComponent implements OnDestroy {
+  filterStoreSub: Subscription;
+  filter: FilterModel;
+  allEpisodesSub: Subscription;
+  episodes: EpisodeModel[];
+  episodeMetricsStoreSub: Subscription;
+  episodeMetrics: any[];
+  episodeTableData: any[];
+  dateRange: any[];
+
+  constructor(public store: Store<any>) {
+
+    this.filterStoreSub = this.store.select(selectFilter).subscribe((newFilter: FilterModel) => {
+      if (newFilter) {
+        if (this.isPodcastChanged(newFilter)) {
+          this.episodeTableData = [];
+        }
+        if (this.isEpisodesChanged(newFilter)) {
+          this.episodeMetrics = filterEpisodeMetrics(newFilter, this.episodeMetrics, 'downloads');
+        }
+        this.filter = newFilter;
+        this.buildEpisodeMetrics();
+      }
+    });
+
+    this.allEpisodesSub = this.store.select(selectEpisodes).subscribe((allEpisodes: EpisodeModel[]) => {
+      const allPodcastEpisodes = filterAllPodcastEpisodes(this.filter, allEpisodes);
+      if (allPodcastEpisodes) {
+        this.episodes = allPodcastEpisodes;
+        this.buildEpisodeMetrics();
+      }
+    });
+
+    this.episodeMetricsStoreSub = this.store.select(selectEpisodeMetrics).subscribe((episodeMetrics: EpisodeMetricsModel[]) => {
+      this.episodeMetrics = filterEpisodeMetrics(this.filter, episodeMetrics, 'downloads');
+      this.buildEpisodeMetrics();
+    });
+  }
+
+  buildEpisodeMetrics() {
+    if (this.episodes && this.episodeMetrics && this.episodeMetrics.length) {
+      this.episodeTableData = this.episodeMetrics
+        .filter((epMetric) => !!this.episodes.find(ep => ep.id === epMetric.id))
```

Comment:
I'm not sure what this filter is for. episodes/allPodcastEpisodes are filtered by the selected podcast, episode metrics are also already filtered by the selected podcast. If anything, it seems more likely that episode metrics would not contain episodes that are in episodes/allPodcastEpisode.